geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegion.java

```diff
@@ -4602,33 +4589,6 @@ void updateNodeToBucketMap(
     }
   }
 
-  /**
-   * @return set of bucket-ids that could not be read from.
-   */
-  private Set<Integer> handleOldNodes(HashMap nodeToBuckets, VersionedObjectList values,
-      ServerConnection servConn) throws IOException {
-    Set<Integer> failures = new HashSet<Integer>();
-    HashMap oldFellas = filterOldMembers(nodeToBuckets);
-    for (Iterator it = oldFellas.entrySet().iterator(); it.hasNext();) {
-      Map.Entry e = (Map.Entry) it.next();
-      InternalDistributedMember member = (InternalDistributedMember) e.getKey();
-      Object bucketInfo = e.getValue();
-
-      HashMap<Integer, HashSet> bucketKeys = null;
-      Set<Integer> buckets = null;
-
-      if (bucketInfo instanceof Set) {
-        buckets = (Set<Integer>) bucketInfo;
-      } else {
-        bucketKeys = (HashMap<Integer, HashSet>) bucketInfo;
-        buckets = bucketKeys.keySet();
-      }
-
-      fetchKeysAndValues(values, servConn, failures, member, bucketKeys, buckets);
-    }
-    return failures;
-  }
-
   void fetchKeysAndValues(VersionedObjectList values, ServerConnection servConn,
```

Comment:
With the removal of `handleOldNodes()`, this method is only used in test code. It could be safely removed along with the test cases that use it. The same is also true of the `getFetchKeysResponse()` and `getValuesForKeys()` methods, which would also now only be used in the same two test cases. The `removeMembersWithSameOrNewerVersion()` method in the `DistributionManager` interface and subclasses is also unused following this change, although in the case of this method, it may be fine to keep it.